### ISSUE TITLE
Test Drive Le Mans (Le Mans 24 Hours) - car shadow missing

### ISSUE THREAD

**Issue by Erick-BR** [flyinghead/flycast#1797]
Platform / OS / Hardware: tested on windows 8.1 and windows 10

Flycast version: lastest version standalone and retroarch with updated cores

Hardware: Ryzen 5 5600; GTX 1660 S; 16GB Ram

Description of the Issue

The shadows of cars, including the player's, do not appear during gameplay. Interestingly, when you pause the game, you can see the shadow underneath the car.

Thank you.

![a](https://github.com/user-attachments/assets/e1090b6f-db03-4de2-ac9d-59d842656357)
![b](https://github.com/user-attachments/assets/c68ca72b-c93e-48f7-bfd1-cf8d3caf5644)

**Comment by flyinghead** [comment]
This looks like a GPU or GPU driver issue. Here is the result with a GTX 1080 (driver version 535.183):
![image](https://github.com/user-attachments/assets/ec4c59ca-4c53-41ca-8f0e-f994032b9109)
Tested with both NTSC and PAL editions.

**Comment by Erick-BR** [comment]
I haven't updated the driver for a long time. I'll do it and test again, but on my Intel HD Graphics 4000 laptop I have the same problem. In the Redream emulator this problem doesn't occur, but of course it's better to play in Retroarch with shaders.

**Comment by flyinghead** [comment]
As I said, this issue is specific to your hardware/software setup. Other games or emulators may or may not experience the same issue depending on the method they use for render.

On Intel you might want to enable `Native Depth Interpolation` but this isn't needed (and not recommended) with Nvidia.
You should also try other graphics API: OpenGL, Vulkan, etc...

**Comment by Erick-BR** [comment]
Still persist on lastest driver 566.36.

Ok, thank you for reply.

**Comment by kihato** [comment]
If you select anything other than per-pixel, the shadow will not be displayed.

**Comment by Erick-BR** [comment]
> If you select anything other than per-pixel, the shadow will not be displayed.

It works with flycast standalone. Do you know where I can find this option on retroarch?

**Comment by kihato** [comment]
Normally shadows are visible in this game regardless of the graphics API or sorting selected, but d618abc appears to have caused a regression.

**Comment by flyinghead** [comment]
Indeed, the mentioned commit is missing some code to be 100% correct. It doesn't happen in per-pixel due to the very different rendering method.
Thanks @kihato.

**Comment by flyinghead** [comment]
Fixed on master
